rotkehlchen/chain/scroll/modules/weth/decoder.py

```diff
@@ -0,0 +1,92 @@
+
+                event.counterparty = self.counterparty
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+                event.notes = f'Wrap {deposited_amount} {self.base_asset.symbol} in {self.wrapped_token.symbol}'  # noqa: E501
+            elif (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == deposited_amount and
+                event.asset == self.wrapped_token
+            ):  # scroll WETH does emit an event on transfer so we can edit the event instead of creating a new one  # noqa: E501
+                event.notes = f'Receive {deposited_amount} {self.wrapped_token.symbol}'
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.counterparty = self.counterparty
+                event.location_label = depositor
+                event.address = context.transaction.to_address
+
+        return DEFAULT_DECODING_OUTPUT
+
+    def _decode_withdrawal_event(self, context: 'DecoderContext') -> 'DecodingOutput':
+        withdrawer = hex_or_bytes_to_address(context.tx_log.topics[1])
+        withdrawn_amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        withdrawn_amount = asset_normalized_value(
+            amount=withdrawn_amount_raw,
+            asset=self.base_asset,
+        )
+        in_event = out_event = None
+        for event in context.decoded_events:
+            if (
+                event.event_type == HistoryEventType.RECEIVE and
+                event.address in (self.wrapped_token.evm_address, withdrawer) and
+                event.balance.amount == withdrawn_amount and
+                event.asset == self.base_asset
+            ):
+                if event.address == withdrawer:
+                    return DEFAULT_DECODING_OUTPUT
+
+                in_event = event
+                event.notes = f'Receive {withdrawn_amount} {self.base_asset.symbol}'
+                event.counterparty = self.counterparty
+            elif (
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE and
+                event.balance.amount == withdrawn_amount and
+                event.asset == self.wrapped_token
+            ):  # scroll WETH does emit an event on transfer so we can edit the event instead of creating a new one  # noqa: E501
+                event.notes = f'Unwrap {withdrawn_amount} {self.wrapped_token.symbol}'
+                event.event_subtype = HistoryEventSubType.RETURN_WRAPPED
+                event.counterparty = self.counterparty
+                event.location_label = withdrawer
+                event.address = context.transaction.to_address
+                out_event = event
+
+        if in_event is None or out_event is None:
+            return DEFAULT_DECODING_OUTPUT
```

Comment:
shouldn't you log an error here? This should never happen with that contract right?